src/coreclr/vm/jitinterface.cpp

```diff
@@ -529,6 +529,41 @@ CEEInfo::ConvToJitSig(
                 sigRet->callConv = (CorInfoCallConv)MetaSig::GetDefaultUnmanagedCallingConvention();
             }
         }
+#if !defined(TARGET_X86)
+        else if ((flags & CONV_TO_JITSIG_FLAGS_CHECK_UNMANAGEDCALLERSONLY) == CONV_TO_JITSIG_FLAGS_CHECK_UNMANAGEDCALLERSONLY
+            && sigRet->callConv == CORINFO_CALLCONV_DEFAULT
+            && context.methodContext && context.methodContext->HasUnmanagedCallersOnlyAttribute())
```

Comment:
If I am reading this right, this will cause an extra custom custom attribute lookup each time a method is JITed, and a new attribute lookup each time a method is inlined.

Would it be better to communicate the unmanaged calling convention for UnmanagedCallersOnly via dedicated channel? E.g. reuse `getUnmanagedCallConv` method for it or introduce a new one. JIT should only call the method for main entrypoint of UnmanagedCallersOnly. 

This should also help with extending the unmanaged calling convention with new values as discussed in https://github.com/dotnet/runtime/pull/39294#issuecomment-735990431

Reply:
I thought that the check for UnmanagedCallersOnly was cached. I just checked and realized it's not.

I'll try doing this via a dedicated channel and see how far I can get.

Reply:
Looks like `m_bFlags2` on `MethodDesc` has room to cache the setting.

Reply:
Boo. Actually that doesn't have room for the calling convention... This is going to make things more complicated.

Reply:
We should not need to cache this. This should be only needed exactly once when the method is JITed.